src/System.CommandLine.Tests/ArgumentTests.cs

```diff
@@ -442,6 +442,52 @@ public void Multiple_command_arguments_can_have_custom_parse_delegates()
                       .Contain("UH UH");
             }
 
+            [Theory]
+            // The two different examples verify that the relative order of the symbols doesn't matter.
+            [InlineData("--option-with-error 123 --depends-on-option-with-error")]
+            [InlineData("--depends-on-option-with-error --option-with-error 123")]
+            public void Custom_parser_can_check_another_option_result_for_custom_errors(string commandLine)
+            {
+                var optionWithError = new CliOption<string>("--option-with-error")
+                {
+                    CustomParser = r =>
+                    {
+                        r.AddError("one");
+                        return r.Tokens[0].Value;
+                    }
+                };
+
+                var optionThatDependsOnOptionWithError = new CliOption<bool>("--depends-on-option-with-error")
+                {
+                    CustomParser = result =>
+                    {
+                        if (result.FindResultFor(optionWithError) is { } optionWithErrorResult)
+                        {
+                            var otherOptionError = optionWithErrorResult.Errors.SingleOrDefault()?.Message;
+
+                            result.AddError(otherOptionError + " " + "two");
```

Comment:
Any reason to not combine the literals?
```suggestion
                            result.AddError(otherOptionError + " two");
```